.github/workflows/test.yaml

```diff
@@ -0,0 +1,76 @@
+name: Test Examples
+
+on:
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  test-tcp-echo:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Create k8s Kind Cluster
+        uses: helm/kind-action@v1
+      - name: Setup
+        working-directory: tcp.echo
+        run: ./setup.sh
+      - name: Execute Test
+        working-directory: tcp.echo
+        run: |
+          set -o pipefail
+          ./test.sh | tee $GITHUB_STEP_SUMMARY
+      - name: Teardown
+        if: always()
+        working-directory: tcp.echo
+        run: ./teardown.sh
+
+  test-tls-echo:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Create k8s Kind Cluster
+        uses: helm/kind-action@v1
+      - name: Setup
+        working-directory: tls.echo
+        run: ./setup.sh
+      - name: Execute Test
+        working-directory: tls.echo
+        run: |
+          set -o pipefail
+          ./test.sh | tee $GITHUB_STEP_SUMMARY
+      - name: Teardown
+        if: always()
+        working-directory: tls.echo
+        run: ./teardown.sh
```

Comment:
If you don't run the `teardown.sh` script, does the kind cluster stay running or does it just delete the whole cluster and everything in it?

Reply:
There is actually a **Post Create k8s Kind Cluster** step that clears up the kind cluster (with whatever it contains) in any case. So in a sense, running the **Teardown** step feels like redundant. However, I still prefer to run it because this way we are also testing for the (unlikely) case if something wrong in the `teardown.sh` script itself, so the whole workflow (setup, test, teardown) is being executed.

Reply:
Sounds good. In that case I don't know if all of teardown scripts have a non 0 exit code if the running containers don't get removed. We will need to make sure all of the teardown scripts can't fail silently since we aren't doing any testing to see if the resources are actually gone or not.

Reply:
We call helm and kubectl in the teardown scripts, it's a reasonable expectation that these tools work as expected and return exit codes properly.